### PR TITLE
Spotlight searches can cause afpd to segfault #56

### PR DESCRIPTION
**Pull request by andy-js**

### CODE REVIEW

File: etc/spotlight/sparql_parser.y [hunk 1]
```diff
@@ -119,6 +119,8 @@ BOOL {
         yyerror("Spotlight queries with logic expressions are disabled");
         YYABORT;
     }
+    if ($1 == NULL || $3 == NULL)
```
Comment (slowfranklin) [inline etc/spotlight/sparql_parser.y]: Can we add a yyerror() here?

File: etc/spotlight/sparql_parser.y [hunk 2]
```diff
@@ -119,6 +119,8 @@ BOOL {
...
+        YYABORT;
```
Comment (slowfranklin) [inline etc/spotlight/sparql_parser.y]: Looks like we should employ the same check above for the expr AND expr case?

Comment (andy-js) [inline etc/spotlight/sparql_parser.y]: I suppose so, though I didn't run into problems there.

File: etc/spotlight/sparql_parser.y (hunk 1 above)
Comment (andy-js) [inline etc/spotlight/sparql_parser.y]: Maybe.  I just copied what the code about did.

**Comment by slowfranklin** [comment]
This got lost somehow. Do you still have the patches somewhere and can create a new merge request?

**Comment by jonassmedegaard** [comment]
The patch is [here](https://salsa.debian.org/netatalk-team/netatalk/-/blob/debian/latest/debian/patches/100~20160429~b30a2ba.patch), in case that's of some use.

**Comment by rdmark** [comment]
Merged to main in https://github.com/Netatalk/netatalk/pull/243

**Pull request closed without merging**